test/common/testlib.py

```diff
@@ -808,6 +810,61 @@ def check_browser_messages(self):
             if log.startswith("error: uncaught"):
                 raise Error(log)
 
+    def check_axe(self, label=None, suffix=""):
+        """Run aXe check on the currently active frame
+
+        The report gets written into an attachment
+        "<label>-axe-{violations,incomplete}.json". If you specify a suffix, it
+        will be appended to the file name, which is useful if you call this
+        more than once within one test.
+        """
+        # only run this on the default OS test, that's enough
+        if os.getenv("TEST_OS") not in [None, testvm.TEST_OS_DEFAULT]:
+            return
+
+        report = self.browser.eval_js("axe.run()", no_trace=True)
+
+        # trim the report
+        def delkeys(dict, *keys):
+            for key in keys:
+                try:
+                    del dict[key]
+                except KeyError:
+                    pass
+        delkeys(report, "passes", "inapplicable", "timestamp")
+
+        for outcome in ["violations", "incomplete"]:
+            for test in report[outcome]:
+                delkeys(test, "tags", "help", "impact")
+
+                # failureSummary in nodes is highly repetitive and long, so summarize it on violation level
+                summaries = set()
+                for result in test["nodes"]:
+                    if "failureSummary" in result:
+                        summaries.add(result["failureSummary"])
+                    delkeys(result, "all", "any", "none", "impact", "failureSummary")
+
+                    # trim containing iframes from targets
+                    if result.get("target", []):
+                        result["target"] = result["target"][-1]
+
+                if summaries:
+                    test["failureSummaries"] = list(summaries)
+
+
+        # write the report
+        if suffix:
+            suffix = "-" + suffix
+        filename = "{0}{1}-axe.json.gz".format(label or self.label(), suffix)
+        with gzip.open(filename, "wb") as f:
+            f.write(json.dumps(report))
+        print("Wrote accessibility report to " + filename)
+        attach(filename)
+
+        # aXe triggers that *shrug*
+        self.allow_journal_messages("received invalid message without channel prefix")
```

Comment:
Oh, very suspicious. Do you have any idea how aXe could trigger sending an invalid message on the web socket?

Reply:
No, sorry, I didn't look at that yet.

Reply:
I'm fine merging it with this, but I'm afraid we'll forget about looking at it. Shall I open an issue about it?